src/main/features/response/guards/guardFactory.ts

```diff
@@ -12,12 +12,16 @@ export class GuardFactory {
     }
   }
 
-  static createAsync (isAllowed: (req: express.Request, res: express.Response) => Promise<boolean>, accessDeniedCallback: (req: express.Request, res: express.Response) => void): express.RequestHandler {
+  static createAsync (isAllowed: (req: express.Request, res: express.Response, next?: express.NextFunction) => Promise<boolean>, accessDeniedCallback: (req: express.Request, res: express.Response) => void): express.RequestHandler {
```

Comment:
I would revert that change, it does not look like you use next handler in `isAllowed`.